src/runmode-af-xdp.c

```diff
@@ -146,6 +148,58 @@ static TmEcode ConfigSetThreads(AFXDPIfaceConfig *aconf, const char *entry_str)
     SCReturnInt(TM_ECODE_OK);
 }
 
+/**
+ * \brief static function to load an AF_XDP program pointed to by the program path
+ *
+ * \param live_dev the interface to add the program to
+ * \param program_path the path to the AF_XDP program to be loaded onto live_dev
+ *
+ */
+static void RunModeLoadAFXDPProg(const char *live_dev, const char *program_path)
+{
+    // we need to register a program
+    struct xdp_program *prog = xdp_program__open_file(program_path, NULL, NULL);
+    int ret = libxdp_get_error(prog);
+    if (ret) {
+        char errmsg[1024];
+        libxdp_strerror(ret, errmsg, sizeof(errmsg));
+        fprintf(stderr, "ERROR: program loading failed: %s\n", errmsg);
+    }
+
+    // attach program in the specified mode
+    unsigned int ifindex = if_nametoindex(live_dev);
+    ret = xdp_program__attach(prog, ifindex, XDP_MODE_NATIVE, 0);
+    if (ret) {
+        printf("ERROR: can't attach\n");
+    }
+
+    free(prog);
+}
+
+/**
+ * \brief function to unload an AF_XDP program
+ *
+ */
+void RunModeAFXDPRemoveProg(void)
+{
+    const char *live_dev = NULL;
+    if (live_dev == NULL)
+        live_dev = LiveGetDeviceName(0);
+    unsigned int ifindex = if_nametoindex(live_dev);
+
+    struct xdp_multiprog *progs = xdp_multiprog__get_from_ifindex(ifindex);
+    if (progs == NULL) {
+        printf("cannot get multiprog");
+    }
+
+    struct xdp_program *prog = xdp_multiprog__main_prog(progs);
+
+    int ret = xdp_program__detach(prog, ifindex, XDP_MODE_NATIVE, 0);
+    if (ret) {
+        printf("ERROR: can't detatch\n");
```

Comment:
We use our own SCLog API for logging messages with adequate log levels. If these are debug messages, I think `SCLogDebug` would work better, if not, maybe `SCLogNotice`?

Reply:
Or, even if this is an unrecoverable error, some other option would be best.